src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -110,7 +113,13 @@ internal ZipArchiveEntry(ZipArchive archive, String entryName)
             _originallyInArchive = false;
 
             _diskNumberStart = 0;
-
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
I thought about splitting the file so we would have a ZipArchiveEntry.Unix and ZipArchiveEntry.Windows but I went with this route since it was more intuitive. Thinking over it more, you're right that it is probably better to not depend on RuntimeInformation if we can avoid i (and we can). I'll split the file.